src/makeInfinityAdder.js

```diff
@@ -4,7 +4,31 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  const adder = function (a) {
+    let sum;
+
+    if (a === undefined) {
+      if (sum === undefined) {
+        sum = 0;
+      }
+
+      return sum;
+    }
+
+    sum = a;
+
+    return (b) => {
+      if (b === undefined) {
+        return sum;
```

Comment:
The sum should be reset to 0 after it is returned. This is to ensure that subsequent calls to 'adder()' after returning the sum start with a fresh sum.